src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -788,6 +810,15 @@ void NonClientIslandWindow::_UpdateFrameMargins() const noexcept
         return FALSE;
     }
 
+    WRL::ComPtr<IUIViewSettingsInterop> interop;
+    Windows::Foundation::GetActivationFactory(WRL::Wrappers::HStringReference(

```

Comment:
@DHowett Please tell me how to convert it from WRL to winrt🙏🙏

Reply:
So, to convert this from WRL to WinRT, all you need to do is.. replace all this code with:

```
auto viewSettingsForWindow = winrt::Windows::UI::ViewManagement::UIViewSettings::GetForWindow();
```

C++/WinRT makes winrt look like modern C++, so you don't need to do any weird smart pointer types or IID management.

Reply:
Oh, but if you have to use an Interop class... then things get weird and annoying. You actually can't use c++/winrt directly for this. lol ☹️

Reply:
Well, using an interop class is the only way I know of. Summoning the legend himself for guidance : @oldnewthing

Reply:
Oh you mean

```cpp
    UIViewSettings viewSettingsForWindow{ nullptr };
    get_activation_factory<IUIViewSettingsInterop>(name_of<UIViewSettings>())->
        GetForWindow(hwnd, guid_of<UIViewSettings>(), put_abi(viewSettingsForWindow));
```

or if you like one-liners

```cpp
    auto viewSettingsForWindow =
        capture<IInspectable>(
            get_activation_factory<IUIViewSettingsInterop>(name_of<UIViewSettings>()),
            &IUIViewSettingsInterop::GetForWindow,
            hwnd).as<UIViewSettings>();
```